manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -542,23 +546,26 @@ func (r *FybrikApplicationReconciler) checkGovernanceActions(configEvaluatorInpu
 	var err error
 	switch configEvaluatorInput.Request.Usage {
 	case taxonomy.WriteFlow:
-		if !req.Context.Requirements.FlowParams.IsNewDataSet {
+		if !req.Context.Requirements.FlowParams.IsNewDataSet ||
+			req.Context.Requirements.FlowParams.ResourceMetadata != nil {
 			// update an existing dataset
 			// query the policy manager whether the operation is allowed
 			reqAction := policymanager.RequestAction{
 				ActionType:         configEvaluatorInput.Request.Usage,
 				Destination:        req.DataDetails.ResourceMetadata.Geography,
 				ProcessingLocation: taxonomy.ProcessingLocation(configEvaluatorInput.Workload.Cluster.Metadata.Region),
 			}
-			req.Actions, err = LookupPolicyDecisions(req.Context.DataSetID, r.PolicyManager, appContext, &reqAction)
+			req.Actions, err = LookupPolicyDecisions(req.Context.DataSetID, req.Context.Requirements.FlowParams.ResourceMetadata,
+				r.PolicyManager, appContext, &reqAction)
 		}
 	case taxonomy.ReadFlow, taxonomy.DeleteFlow:
 		reqAction := policymanager.RequestAction{
 			ActionType:         configEvaluatorInput.Request.Usage,
 			Destination:        configEvaluatorInput.Workload.Cluster.Metadata.Region,
 			ProcessingLocation: taxonomy.ProcessingLocation(configEvaluatorInput.Workload.Cluster.Metadata.Region),
 		}
-		req.Actions, err = LookupPolicyDecisions(req.Context.DataSetID, r.PolicyManager, appContext, &reqAction)
+		req.Actions, err = LookupPolicyDecisions(req.Context.DataSetID, req.Context.Requirements.FlowParams.ResourceMetadata,
+			r.PolicyManager, appContext, &reqAction)
```

Comment:
What is the meaning of taking metadata not from the catalog for read/delete scenarios?

Reply:
we will revisit this part. ResourceMetadata from flow requirements should be taken into account only when isNewDataSet is true

Reply:
Even not then. You don't know the destination at this point, so what will the WRITE query be about?

Reply:
not sure if that answers the question but we have the tags so we can get policies related to it.

Reply:
@shlomitk1 as you suggested the metadata is now passed to the policy compiler and the call to the data catalog from policy compiler is removed.